comment_scan/chatter.rb

```diff
@@ -57,33 +58,48 @@ def add_fall_through_reply_action(action, args_to_pass=nil)
     end
 
     def mention_received(room_id, message)
+        #Grab/create/update chat user
+        chat_user = ChatUser.find_or_create_by(user_id: message.hash['user_id'])
+        chat_user.update(name: message.hash['user_name'])
+
         @mention_actions.each do |action, payload|
-            action.call(*payload, message.id, room_id, message.body)
+            #Run at most one mention action successfully
+            return true if action.call(*payload, message.id, chat_user, room_id, message.body)
         end
+        return false
     end
 
     def reply_received(room_id, message)
         return unless message.hash.include? 'parent_id'
 
+        #Grab/create/update chat user
+        chat_user = ChatUser.find_or_create_by(user_id: message.hash['user_id'])
+        chat_user.update(name: message.hash['user_name'])
+
         reply_args = message.body.downcase.split(' ').drop(1) #Remove the reply portion
-        return if reply_args.empty? #No args
+        return false if reply_args.empty? #No args
         reply_command = reply_args[0]
         reply_args = reply_args.drop(1) #drop the command
 
         if @reply_actions.key?(reply_command)
             begin
-                @reply_actions[reply_command].each do |action, args_to_pass|
-                    action.call(*args_to_pass, message.id, message.hash['parent_id'], room_id, *reply_args)
+                reply_responses = @reply_actions[reply_command].map do |action, args_to_pass|
+                    action.call(*args_to_pass, message.id, message.hash['parent_id'], chat_user, room_id, *reply_args)
                 end
+                return reply_responses.any? #return true if any responses got run
             rescue ArgumentError => e
                 say("Invalid number of arguments for '#{reply_command[0]}' command.", room_id)
                 @logger.warn e
-                #TODO: Would be cool to have some help text print here. Maybe we could pass it when we do add_command_action?
+                #TODO: Would be cool to have some help text print here. Maybe we could pass it when we do add_reply_action?
             rescue Exception => e
                 say("Got exception ```#{e}``` processing your response", room_id)
             end
+            return true
         else
-            @fall_through_actions.each { |action, payload| action.call(*payload, message.id, message.hash['parent_id'], room_id, *reply_args)}
+            @fall_through_actions.each do |action, payload|
+                action.call(*payload, message.id, message.hash['parent_id'], chat_user, room_id, *reply_args)
+            end
+            return false
```

Comment:
All of these returns smell funny. I'll need to look closer later, but it feels simplifiable

Reply:
Agreed. This was the simplest way I could think of (and what would work in C++ - which usually means it's the wrong choice for Ruby in my experience :p). But it's working for now. If you can think of something, let me know!

Reply:
Looking at this again, I think I'd make it feel a bit more guard-ey, so
```
if !@reply_actions.key?(reply_command)
  # fall through actions
  return false
end

begin
  whatever.any? {}
rescue
  # ...
  true
rescue
  # ...
  true
end
```